tests/phpunit/integration/Modules/AdsTest.php

```diff
@@ -21,46 +23,97 @@
+		$this->ads     = new Ads( $this->context );
+	}
+
 	public function test_magic_methods() {
-		$ads = new Ads( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+		$this->assertEquals( 'ads', $this->ads->slug );
+		$this->assertEquals( 'Ads', $this->ads->name );
+		$this->assertEquals( 'https://google.com/ads', $this->ads->homepage );
+	}
 
-		$this->assertEquals( 'ads', $ads->slug );
-		$this->assertEquals( 'Ads', $ads->name );
-		$this->assertEquals( 'https://google.com/ads', $ads->homepage );
+	public function test_register() {
+		remove_all_actions( 'template_redirect' );
+		remove_all_filters( 'googlesitekit_inline_modules_data' );
+
+		$this->ads->register();
+
+		$this->assertTrue( has_action( 'template_redirect' ) );
+		$this->assertTrue( has_filter( 'googlesitekit_inline_modules_data' ) );
 	}
 
 	public function test_is_connected_when_ads_conversion_id_is_set() {
-		$ads = new Ads( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+		$this->assertFalse( $this->ads->is_connected() );
 
-		$this->assertFalse( $ads->is_connected() );
+		$this->ads->get_settings()->set( array( 'conversionID' => 'AW-123456789' ) );
 
-		$ads->get_settings()->set( array( 'conversionID' => 'AW-123456789' ) );
+		$this->assertTrue( $this->ads->is_connected() );
+	}
 
-		$this->assertTrue( $ads->is_connected() );
+	public function test_inline_data_initial_state__module_not_connected() {
+		$inline_modules_data = apply_filters( 'googlesitekit_inline_modules_data', array() );
+
+		$this->assertArrayNotHasKey( 'ads', $inline_modules_data );
 	}
 
-	public function test_settings_reset_on_deactivation() {
-		$ads = new Ads( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+	public function test_inline_custom_dimension_data_initial_state__module_connected() {
+		$this->ads->register();
 
-		$ads->get_settings()->set( array( 'conversionID' => 'AW-123456789' ) );
+		// Ensure the module is connected.
+		$options = new Options( $this->context );
+		$options->set(
+			Settings::OPTION,
+			array(
+				'conversionID' => 'AW-12345',
+			)
+		);
 
-		$ads->on_deactivation();
+		$inline_modules_data = apply_filters( 'googlesitekit_inline_modules_data', array() );
 
-		$ads_settings = $ads->get_settings()->get();
+		$this->assertEquals(
+			array(
+				'supportedConversionEvents' => array(),
+			),
+			$inline_modules_data['ads']
+		);
+	}
+
+	public function test_settings_reset_on_deactivation() {
+		$this->ads->get_settings()->set( array( 'conversionID' => 'AW-123456789' ) );
+
+		$this->ads->on_deactivation();
+
+		$ads_settings = $this->ads->get_settings()->get();
 
 		$this->assertFalse( $ads_settings );
 	}
 
 	public function test_get_debug_fields() {
-		$ads = new Ads( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
-
-		$ads->get_settings()->set( array( 'conversionID' => 'AW-123456789' ) );
+		$this->ads->get_settings()->set( array( 'conversionID' => 'AW-123456789' ) );
 
 		$this->assertEqualSets(
 			array(
 				'conversion_tracking_id',
```

Comment:
This looks like the branch is not up to date with the latest since its debug fields should all have the `ads_` prefix 🤔

Reply:
This seems to be latest update, it was named adsConversionID in the past, but was renamed

Reply:
The setting itself is `conversionID` which is correct.
The debug field (Site Health) should be `ads_conversion_tracking_id` – see https://github.com/google/site-kit-wp/blob/cc4f31ffd6ed48ca1a4c4464d8b4551b76500274/tests/phpunit/integration/Modules/AdsTest.php#L57-L76 which is the current develop, and correct.

So it seems this PR is 30+ commits behind develop so it probably just needs merging the latest.